src/telliot_feeds/sources/mimicry.py

```diff
@@ -0,0 +1,331 @@
+                values.append(sale.price)
+            else:
+                values.append(int(transaction_history.floor_price))
+
+            for other_sale in transaction_history.transactions:
+                if other_sale.item_id == sale.item_id:
+                    transaction_history.transactions.remove(other_sale)
+
+        # return(sum(values))
+        return sum(values)
+
+    async def request_historical_sales_data(self, contract: str, all: bool = True) -> Optional[TransactionList]:
+        """Requests historical sales
+         data of the selected collection.
+         Data retrieved from Reservoir.
+
+        Agruments:
+            all (bool): if True, see all data for the selected collection (if False, only 12 months)
+
+        Returns:
+            TransactionList: formatted historical sales data of a collection retrieved from Reservoir
+
+        """
+        url = f"https://api.reservoir.tools/sales/v4?contract={contract}"
+        headers = {"accept": "*/*", "x-api-key": "demo-api-key"}
+        with requests.Session() as s:
+            s.mount("https://", adapter)
+            if not all:
+                one_year_ago = datetime.now() - timedelta(days=365)
+                start_timestamp = int(one_year_ago.timestamp())
+                url += f"&startTimestamp={start_timestamp}"
+            try:
+                request = s.get(url, timeout=0.5, headers=headers)
+            except requests.exceptions.RequestException as e:
+                logger.error(f"Reservoir API error: {e}")
+                return None
+
+            except requests.exceptions.Timeout as e:
+                logger.error(f"Reservoir API timed out: {e}")
+                return None
+
+            if not request.ok:
+                logger.error(f"Reservoir API error: {request.text}")
+                return None
+
+            tx_list = TransactionList()
+
+            for sale in request.json()["sales"]:
+
+                try:
+                    price = sale["price"]["amount"]["usd"]
+                    item_id = sale["token"]["tokenId"]
+                    timestamp = sale["timestamp"]
+
+                    tx = Transaction(price=price, item_id=item_id, timestamp=timestamp)
+
+                    tx_list.transactions.append(tx)
+                except KeyError as e:
+                    logger.warn("Mimicry: Reservoir Sales API KeyError: " + str(e))
+
+            if all:
+                url = (
+                    "https://api.reservoir.tools/oracle/collections/floor-ask/v4?kind="
+                    f"spot&currency=0x0000000000000000000000000000000000000000&twapSeconds=0&collection={contract}"
+                )
+                headers = {"accept": "*/*", "x-api-key": "demo-api-key"}
+
+                try:
+                    request = s.get(url, timeout=0.5, headers=headers)
+                except requests.exceptions.RequestException as e:
+                    logger.error(f"Reservoir API error: {e}")
```

Comment:
Might want to check your logging messages in this file and make 'em as specific as possible. There's some repeat ones, which will make debugging harder in the future